edfi-repo-auditor/edfi_repo_auditor/checklist.py

```diff
@@ -23,33 +25,135 @@
                       "DELETES_HEAD",
                       "USES_SQUASH",
                       "LICENSE_INFORMATION",
-                      "DEPENDABOT_HAS_ALERTS",
+                      "DEPENDABOT_ENABLED",
+                      "DEPENDABOT_ALERTS",
                       "README",
                       "CONTRIBUTORS",
                       "NOTICES",
-                      "LICENSE"])
+                      "LICENSE"
+                    ])
 
 CHECKLIST = options(
-              HAS_ACTIONS="Has Actions",
-              CODEQL="Uses CodeQL",
-              APPROVED_ACTIONS="Uses only approved GitHub Actions",
-              TEST_REPORTER="Uses Test Reporter",
-              UNIT_TESTS="Has Unit Tests",
-              LINTER="Has Linter",
-              SIGNED_COMMITS="Requires Signed commits",
-              CODE_REVIEW="Requires Code review",
-              REQUIRES_PR="Requires PR",
-              ADMIN_PR="Admin cannot bypass PR",
-              WIKI="Has Wiki Enabled",
-              ISSUES="Has Issues Enabled",
-              PROJECTS="Has Projects Enabled",
-              DISCUSSIONS="Has Discussions",
-              DELETES_HEAD="Deletes head branch",
-              USES_SQUASH="Uses Squash Merge",
-              LICENSE_INFORMATION="Has License Information",
-              DEPENDABOT_HAS_ALERTS="Has Dependabot alerts",
-              README="README.md",
-              CONTRIBUTORS="CONTRIBUTORS.md",
-              NOTICES="NOTICES.md",
-              LICENSE="LICENSE"
-              )
+              HAS_ACTIONS={
+                "description": "Has Actions",
+                "success": DEFAULT_SUCCESS_MESSAGE,
+                "fail": "Repo is not using GH Actions"
+              },
+              CODEQL={
+                "description": "Uses CodeQL",
+                "success": DEFAULT_SUCCESS_MESSAGE,
+                "fail": "CodeQL not found"
+              },
+              APPROVED_ACTIONS={
+                "description": "Uses only approved GitHub Actions",
+                "success": DEFAULT_SUCCESS_MESSAGE,
+                "fail": "Consider using only approved GH Actions"
```

Comment:
Note to self: in a future revision, change so that the repo "fails" if this test fails, even if everything else is perfect. Not part of the requirements today though, so no change right now.